deployment/terraform/create.go

```diff
@@ -312,15 +312,22 @@ func (t *Terraform) createAdminUser(extAgent *ssh.ExtAgent, output *Output) erro
 func (t *Terraform) updateAppConfig(ip string, sshc *ssh.Client, output *Output) error {
 	var clusterDSN, driverName string
 	var readerDSN []string
-	switch t.config.DBInstanceEngine {
-	case "aurora-postgresql":
-		clusterDSN = "postgres://" + t.config.DBUserName + ":" + t.config.DBPassword + "@" + output.DBCluster.ClusterEndpoint + "/" + t.config.ClusterName + "db?sslmode=disable"
-		readerDSN = []string{"postgres://" + t.config.DBUserName + ":" + t.config.DBPassword + "@" + output.DBCluster.ReaderEndpoint + "/" + t.config.ClusterName + "db?sslmode=disable"}
-		driverName = "postgres"
-	case "aurora-mysql":
-		clusterDSN = t.config.DBUserName + ":" + t.config.DBPassword + "@tcp(" + output.DBCluster.ClusterEndpoint + ")/" + t.config.ClusterName + "db?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s"
-		readerDSN = []string{t.config.DBUserName + ":" + t.config.DBPassword + "@tcp(" + output.DBCluster.ReaderEndpoint + ")/" + t.config.ClusterName + "db?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s"}
-		driverName = "mysql"
+
+	clusterDSN = t.config.ExternalDBSettings.DataSource
+	readerDSN = t.config.ExternalDBSettings.DataSourceReplicas
+	driverName = t.config.ExternalDBSettings.DriverName
+
+	if t.config.TerraformDBSettings.InstanceCount > 0 {
```

Comment:
Let's refactor this to a helper method to check the terraform output struct instead like we do below. Something like `if output.HasDB() {`

Reply:
I think now is done, but I'm not 100% sure if the `DBCluster.ClusterEndpoint` is going to be an empty string on the output.

Reply:
Can we verify that? You don't need to go through the entire rigmarole of setting up app servers and everything. Just comment out everything after `output, err := t.Output()` in (*Terraform).Create, create a deployment, and inspect the tfstate file.

Reply:
Verified, it works well :)